test/functional/url_parser_tests.js

```diff
@@ -426,7 +426,27 @@ exports['Parse mongodb://localhost/db?w=-1'] = {
 /**
  * @ignore
  */
-exports['Throw on unsuported options'] = {
+exports['Should be able to parse mongodb://localhost/?compressors=snappy,zlib&zlibCompressionLevel=-1'] = {
```

Comment:
What about these cases:
  - one only compressor specified
  - `zlibCompressionLevel` specified with no compressors
  - `zlibCompressionLevel` specified with `snappy` compression

I know we discussed that the last two are probably no-ops (due to spec), but let's put our assumptions to paper here.